### DISCUSSION TITLE
possible feature, cp: respect .gitignore

### DISCUSSION THREAD

**Discussion by PThorpe92** [uutils/coreutils#4973] [Ideas]
This feature was suggested in another discussion. Is adding an arg/flag to the cp command to enable respect of a .gitignore file something that the project would be interested in adding? If this is the case, I would like to attempt to implement this, but would have some questions about specifics of desired implementation.

**Comment by tertsdiepraam** [comment]
The question is what you want `cp` to do here. `cp` just copies the files it explicitly gets. That does not include globs. For example: say I have `a.txt` and `b.txt` and `a.txt` is in gitignore. Then `cp *.txt` will be expanded by the shell to `cp a.txt b.txt`. Hence there's no way for us to distinguish between `cp a.txt b.txt` and `cp *.txt`. Even with an option to ignore gitignore files, it seems weird to ignore files that are explicitly passed to `cp`. Another question is whether you want it to filter recursively. So if I have a folder with an ignored file and I `cp` that folder, should the ignored file be in the copy?

**Reply by PThorpe92** [reply]
This was an issue I had to deal with in another project. Because it's disabled by default, I feel the recursive option would be the main use case, e.g. copying a folder/.git repo, and wanting the .gitignored files to not be copied along with it. Much like when you push remote.
This and glob (cp * ./some/dir) would seem to me to be the most likely use-cases. It may not be a good fit though, just thought I would get some opinions.

**Comment by anoraktrend** [comment]
I think your request might be better suited for a shell to handle rather than cp.
